trin-validation/src/accumulator.rs

```diff
@@ -155,37 +155,58 @@ impl PreMergeAccumulator {
         }
 
         // Create a merkle tree from epoch accumulator.
-        // To construct a valid proof for the header hash, we add both the hash and the total
-        // difficulty as individual leaves for each header record. This will ensure that the
-        // total difficulty is included as the first element in the proof.
+        // To construct a valid proof for the header hash, we add a leaf of
+        // hash(header, total difficulty) for each header record at a depth of 13.
+        // This must be done to support generating valid proofs for partial
+        // epochs, as opposed to adding the header and total difficulty as individual
+        // leaves on a tree of depth 14.
+        //
+        // Then we re-insert the total difficulty as the first element
+        // in the proof to be able to prove the header hash.
         let mut leaves = vec![];
+        let mut header_difficulty = B256::ZERO;
```

Comment:
I think you can extract header difficulty directly at this point:
```rust
let header_difficulty = B256::from(header_record.total_difficulty.to_le_bytes());
```
And you can also do it for header_record_hash:
```rust
let mut header_record_hash = B256::from(eth2_hashing::hash32_concat(
    header_record.block_hash.as_slice(),
    current_difficulty.as_slice(),
);
```

With above, you can even remove the loop:
```rust
let leaves = epoch_acc
    .iter()
    .map(|record| {
        B256::from(eth2_hashing::hash32_concat(
            record.block_hash.as_slice(),
            record.total_difficulty.as_le_slice(),
        ))
    })
    .collect::<Vec<_>>();
```
(or you can keep it as a loop, but much simpler)